app/controllers/comment_controller.php

```diff
@@ -63,26 +80,102 @@ public function edit()
         $title = $thread->title;
         $page = Param::get('page_next', 'edit');
 
-
         switch ($page) {
-        case 'edit':
-            break;
-        case 'edit_end':
-           $comment->body = Param::get('body');
-
-            try {
-                $comment->edit($comment_id);
-                redirect("/comment/view?thread_id={$_SESSION['thread_id']}");
-            } catch (ValidationException $e) {
-                $page = 'edit';
-            }
-            break;  
-        default:
-            throw new NotFoundException("{$page} not found");
-            break;
+            case 'edit':
+                break;
+            case 'edit_end':
+               $comment->body = Param::get('body');
+
+                try {
+                    $comment->edit($comment_id);
+                    redirect("/comment/view?thread_id={$_SESSION['thread_id']}");
+                } catch (ValidationException $e) {
+                    $page = 'edit';
+                }
+                break;  
+            default:
+                throw new NotFoundException("{$page} not found");
+                break;
         }
 
         $this->set(get_defined_vars());
         $this->render($page);
     }
+
+    public function delete()
+    {
+        $comments = new Comment();
+
+        $comment_id = Param::get('comment_id');
+
+        $comments->delete($comment_id);
```

Comment:
Please check if the comment is existing first before calling the `delete()` method. :bow:
This is to get the instance of `Comment` po, rather than initializing it.. ie.
```php
try {
    $comment = Comment::get($comment_id);
    $comment->delete();
} catch (exception) {
    // render page if comment is not found
}
```

Also, please add the proper restriction po, anyone who can access `commend/delete?comment_id=` can delete any comment, even if they are not logged in or the comment is not created by them. :bow: